Marlin/Marlin_main.cpp

```diff
@@ -1095,10 +1095,7 @@ static void set_bed_level_equation_lsq(double *plane_equation_coefficients)
 //    corrected_position.debug("position after");
     current_position[X_AXIS] = corrected_position.x;
     current_position[Y_AXIS] = corrected_position.y;
-    current_position[Z_AXIS] = corrected_position.z;
-
-    // put the bed at 0 so we don't go below it.
-    current_position[Z_AXIS] = zprobe_zoffset; // in the lsq we reach here after raising the extruder due to the loop structure
+    current_position[Z_AXIS] = corrected_position.z + zprobe_zoffset;
```

Comment:
Still waiting to find out whether this is good or bad. @nophead leaning towards bad…